example/lib/src/ble/ble_device_connector.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:async';
+
+import 'package:flutter_reactive_ble/flutter_reactive_ble.dart';
+import 'package:flutter_reactive_ble_example/src/ble/reactive_state.dart';
+
+class BleDeviceConnector extends ReactiveState<ConnectionStateUpdate> {
+  BleDeviceConnector(this._ble);
+
+  final FlutterReactiveBle _ble;
+
+  @override
+  Stream<ConnectionStateUpdate> get state => _deviceConnectionController.stream;
+
+  final _deviceConnectionController = StreamController<ConnectionStateUpdate>();
+
+  StreamSubscription<ConnectionStateUpdate> _connection;
+
+  Future<void> connect(String deviceId) async {
+    if (_connection != null) {
+      await _connection.cancel();
+    }
+    _connection = _ble.connectToDevice(id: deviceId).listen(
+          _deviceConnectionController.add,
+        );
+  }
+
+  Future<void> disconnect(String deviceId) async {
+    if (_connection != null) {
+      try {
+        await _connection.cancel();
+      } on Exception catch (e, _) {
+        print("Error disconnecting from a device: $e");
+      } finally {
```

Comment:
@werediver  maybe we should document this (in readme or faq?) since it can be confusing at first that after disconnect you do not get a disconnected event.